templates/nginx-metrics.conf.j2

```diff
@@ -0,0 +1,39 @@
+{% if restund_metrics_enabled %}
+server {
+    listen {{ restund_metrics_listen_port }} ssl;
+    server_name {{ certbot_domain }};
+
+    ssl_certificate /etc/letsencrypt/live/{{ certbot_domain }}/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/{{ certbot_domain }}/privkey.pem;
+
+    # intermediate configuration as per https://ssl-config.mozilla.org/#server=nginx
+    ssl_session_timeout 1d;
+    ssl_session_cache shared:MetricsSSL:10m;  # about 40000 sessions
+    ssl_session_tickets off;
+
+    ssl_protocols TLSv1.2 TLSv1.3;
+    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+    ssl_prefer_server_ciphers off;
+
+    {% if restund_metrics_client_certificates_enabled %}
+    ssl_client_certificate /etc/nginx/ssl/certs/ca.crt;
+    ssl_verify_client      on;
+    {% endif %}
+
+    location /metrics/node {
+        proxy_pass http://127.0.0.1:9100/metrics;
+        proxy_http_version 1.1;
+    }
+
+    location /metrics/restund {
+        proxy_pass http://127.0.0.1:9200/metrics;
+        proxy_http_version 1.1;
+    }
+
+    location /metrics/process {
+        proxy_pass http://127.0.0.1:9256/metrics;
+        proxy_http_version 1.1;
+    }
+
+}
+{% endif %}
```

Comment:
I'm probably overlooking sth here, but wouldn't this create an empty file combined with

```
- name: template nginx config files (locally)
  template: ...
  with_items:
    ...
    - { src: nginx-metrics.conf.j2, dest: "{{ nginx_localhost_config_dir }}/http/metrics.conf" }
    ...
...
```

If that is the case, maybe a `template` task conditioned by `restund_metrics_enabled` as an alternative would prevent this.

Reply:
Yes, an empty file is created if it's disabled. I added an else block with a comment in https://github.com/wireapp/ansible-restund/pull/13/commits/074a6f41a8f310d0f8a918df09cff5162ba26c0b

While entirely not creating the file would be possible with an extra task with a condition, removing it from a server when the value changes from true to false on an existing server is not something that's supported by the external role in this way (and indeed a logic that can't easily be implemented with the `copy` and `template` modules).

My plan is to add the node/process/rex exporters to this role in follow-up PRs, and once that is done, set the default value of restund_metrics_enabled to true.

Reply:
Ah, yes due to how the external nginx role is being integrated here. Makes sense form a technical point of view. And an additional `file: { state = absent when = not restund_metrics_enabled }` for that template output is an even uglier workaround. Are you still convinced about that nginx role ;)